### DISCUSSION TITLE
Several Types of Auth With Same ID/DNS entry

### DISCUSSION THREAD

**Discussion by anav000** [strongswan/strongswan#2145] [Q&A] [answered]
Hello,

I'm reaching out to inquire about handling multiple types of authentication on the server side using the same ID (as specified in the conf/certs to match the DNS entry).

Currently, I have connections A and B set up on the same server, both functioning correctly. However, it seems that when attempting to connect to connection B from the client side, the server automatically selects connection A, which leads to authentication errors due to the differing authentication types.

Any suggestions on how to ensure the server selects the correct connection?

to illustrate: server's swanctl.conf
`
connections {
      conn-A {
            .....
            local_addrs = my.server.addrs.com
            .........
            
            local {
                  auth = X
                  ......
            }
            
            remote {
                  auth = Z
                  .....
            }
            
            children {
                  child-A {
                       .....
                  }
            }
      }
      conn-B {
            .....
            local_addrs = my.server.addrs.com # same
            .........
            
            local {
                  auth = X
                  ......
            }
            
            remote {
                  auth = Y # different
                  .....
            }
            
            children {
                  child-B {
                       .....
                  }
            }
      }
}

........
`

**Comment by tobiasbrunner** [comment, marked as answer]
> Any suggestions on how to ensure the server selects the correct connection?

The config is basically selected based on the IP addresses, the IKE proposals (only to weed out incompatible connections, no best match) and the identities (local and remote). So if two configs are matching exactly the same based on these properties, the first one found will be used initially. The others are kept as possible candidates, so there might be a switch to another config based on some constraints, of which the authentication type is one. You can check the log to see what's going on exactly in your case (set the log level for _cfg_ to 2 for more details).

**Reply by anav000** [reply]
Thank you for your response. I attempted to reconnect to check the logs and it seems to be working now. I'm not entirely sure what changes were made on the server/client side to make it work, but I'm glad it's resolved.

Regarding the eap-radius plugin, would it be better to create a separate topic for that discussion?

Best regards

**Reply by tobiasbrunner** [reply]
> Regarding the eap-radius plugin, would it be better to create a separate topic for that discussion?

Not sure what you are referring to, but probably yes.
